File: JIWON_CODE/052 최대공약수와 최소공배수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer;
    int num = m;
    
    for(int i=n; i>=1; i--){
        if(n % i == 0 && m % i == 0){
            answer.push_back(i);
            break;
        }
    }
    
    while(true){
        if(num % n == 0){
            answer.push_back(num);
            break;
        }
        num += m;
    }
    
    return answer;
}

File: lesunny/052_최대공약수와최소공배수.java
class Solution {
    public int gcd (int n, int m){
        if(m==0)
            return n;
        else
            return gcd(m, n%m);
    }
    
    public int lcm (int n, int m, int gcd){
        return n*m/gcd;
    }
    
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        answer[0] = gcd(n,m);
        answer[1] = lcm(n,m,answer[0]);
        
        return answer;
    }
}
